src/systems/userinput/devices/vive-controller.js

```diff
@@ -17,28 +17,55 @@ export class ViveControllerDevice {
     navigator.getVRDisplays();
     this.gamepad = gamepad;
 
-    if (this.gamepad.id === "HTC Vive Focus Plus Controller") {
+    if (this.gamepad.id === "OpenVR Cosmos") {
+      RAY_ROTATION.setFromAxisAngle(new THREE.Vector3(1, 0, 0), -Math.PI / 12);
+      this.buttonMap = [
+        { name: "trigger", buttonId: 0 },
+        { name: "grip", buttonId: 1 },
+        { name: "primary", buttonId: 2 },
+        { name: "secondary", buttonId: 3 },
+        { name: "joystick", buttonId: 4 },
+        { name: "bumper", buttonId: 5 }
+      ];
+      this.axisMap = [{ name: "joyX", axisId: 0 }, { name: "joyY", axisId: 1 }];
+    } else if (this.gamepad.id === "HTC Vive Focus Plus Controller") {
       RAY_ROTATION.setFromAxisAngle(new THREE.Vector3(1, 0, 0), Math.PI / 15);
       this.buttonMap = [
         { name: "touchpad", buttonId: 0 },
         { name: "trigger", buttonId: 1 },
         { name: "grip", buttonId: 2 }
       ];
+      this.axisMap = [{ name: "touchX", axisId: 0 }, { name: "touchY", axisId: 1 }];
+    } else if (this.gamepad.axes.length === 4) {
+      RAY_ROTATION.setFromAxisAngle(new THREE.Vector3(1, 0, 0), -Math.PI / 12);
+      this.buttonMap = [
+        { name: "touchpad", buttonId: 0 },
+        { name: "trigger", buttonId: 1 },
+        { name: "grip", buttonId: 2 },
+        { name: "primary", buttonId: 3 },
```

Comment:
A bit unfortunate to have to remember which button is "primary" on the different controllers, but doesn't seem like a big deal